### PR TITLE
refactor(nodebuilder): Move parse methods and flag definitions to nodebuilder subpackages

### LINKED ISSUE #1126: refactor(nodebuilder): Move parse functions + flag defs to the respective modules
https://github.com/celestiaorg/celestia-node/pull/997#discussion_r970083829

Blocked by #997

### PR DESCRIPTION
**Pull request by distractedm1nd**
Closes #1126

### CODE REVIEW

**Review by vgonkivs — APPROVED** [review]

**Review by Wondertan — CHANGES_REQUESTED** [review]
Actually, we should make one nice thing here. Every flag name is prepended with the name of a module, so let's extract module name strings we pass to fx.Module into constants and reuse them as categories for flag names.

**Comment by distractedm1nd** [comment]
```golang
var (
	ipFlag   = moduleName + ".ip"
	rpcFlag  = moduleName + ".rpc.port"
	grpcFlag = moduleName + ".grpc.port"
)
```
Ok so I tried it out @Wondertan:

I don't think this is more readable, and then there is for example the "keyring.accname" in the state package. I'm not convinced extracting this into a const is better yet

File: nodebuilder/header/flags.go [hunk 1]
```diff
@@ -27,16 +24,16 @@ func HeadersFlags() *flag.FlagSet {
 	return flags
 }
 
-// ParseHeadersFlags parses Header package flags from the given cmd and applies values to Env.
-func ParseHeadersFlags(ctx context.Context, cmd *cobra.Command, cfg *nodebuilder.Config) (context.Context, error) {
-	if ctx, err := ParseTrustedHashFlags(ctx, cmd, cfg); err != nil {
-		return ctx, err
+// ParseFlags parses Header package flags from the given cmd and saves them to the passed config.
```
Comment (renaynay) [inline nodebuilder/header/flags.go:27]:
```suggestion
// ParseFlags parses Header package flags from the given cmd and applies them to the passed config.
```

we don't necessarily *save* the modified conf, only on Init

File: nodebuilder/header/flags.go [hunk 2]
```diff
@@ -50,28 +47,24 @@ func TrustedPeersFlags() *flag.FlagSet {
 	return flags
 }
 
-// ParseTrustedPeerFlags parses Header package flags from the given cmd and applies values to Env.
+// ParseTrustedPeerFlags parses Header package flags from the given cmd and saves them to the passed config.
```
Comment (renaynay) [inline nodebuilder/header/flags.go:50]: same thing here, just leave it as `applies`

**Review by renaynay — DISMISSED** [review]

File: nodebuilder/header/flags.go (hunk 2 above)
Comment (distractedm1nd) [reply nodebuilder/header/flags.go]: done

**Comment by renaynay** [comment]
also generally, if we have ideas like this that get spawned from a PR, unless they're like one-line fixes, we should make a habit of sticking them in an issue and linking the PR inspiring it rather than increasing scope of PR

**Review by walldiss — APPROVED** [review]

**Review by Wondertan — APPROVED** [review]
> I don't think we should do this now as it is yet another breaking change

It is fine to group multiple breaking changes together.

> We can make an issue for it to do it down the line but i wouldn't hold this PR up for it and i think our current naming is okay.

Agreed

**Review by renaynay — APPROVED** [review]

**Pull request merged**